core-bundle/src/Csp/WysiwygStyleProcessor.php

```diff
@@ -61,6 +61,6 @@ public function extractStyles(string $htmlFragment): array
 
     public static function prepareRegex(string $regex): string
     {
-        return '/^'.$regex.'$/';
+        return '(^(?:'.$regex.')$)i';
```

Comment:
Makes sure regexs like `a|b|c` match beginning and end for all three values. Also added a test for this.